modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -147,17 +150,140 @@ class PolygonMeshGeneratorBase : public MeshGenerator
                    const bool quad_center_elements = false,
                    const boundary_id_type boundary_id_shift = 0);
 
+  /**
+   * Creates a mesh of a general slice that with a triangle shape with circular regions on one of
+   * its vertex.
+   * @param ring_radii radii of the ring regions
+   * @param ring_layers numbers of radial intervals of the ring regions
+   * @param ring_radial_biases values used for radial meshing biasing in ring regions
+   * @param ring_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the ring regions
+   * @param ring_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the ring regions
+   * @param ducts_center_dist distance parameters of the duct regions
+   * @param ducts_layers numbers of radial intervals of the duct regions
+   * @param duct_radial_biases values used for radial meshing biasing in duct regions
+   * @param duct_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the duct regions
+   * @param duct_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the duct regions
+   * @param primary_side_length length of the first side that involves the ring center vertex
+   * @param secondary_side_length length of the second side that involves the ring center vertex
```

Comment:
so this is the radius of the first two "rings" at the center of the polygon?

Reply:
Previously, the slice has to be an isosceles triangle. Here, a general triangle (with ring regions near one of its vertices) can be generated. Thus, we need to define two side lengths (primary and secondary) instead of one (deduced from `pitch`).

Reply:
makes sense.
What determines first / second? Is this in increasing azimuthal angle?

Reply:
if `rotation_angle` is 0, the first side is parallel to y-axis; the second side direction can be obtained by clockwise rotating the first side by `azimuthal_angle`.  Will add these.